### PR TITLE
Azure/AFD-premium-default-rule-set

### PR DESCRIPTION
**Pull request by alphadev4**

### CODE REVIEW

File: plugins/azure/frontdoor/frontDoorWafDefaultRuleSet.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+                premiumWAF = true;
+                var ruleSet = (policy.managedRules &&
+                    policy.managedRules.managedRuleSets &&
+                    policy.managedRules.managedRuleSets.find(set => set.ruleSetType && set.ruleSetType.toLowerCase() === 'microsoft_defaultruleset')) || {};
+
+                var ruleSetVersion = ruleSet.ruleSetVersion ? parseFloat(ruleSet.ruleSetVersion) : '';
+                var ruleSetAction = ruleSet.ruleSetAction ? ruleSet.ruleSetAction.toLowerCase() : '';
+
+                if (ruleSetVersion >= 2.1) {
```
Comment (mehakseedat63) [inline plugins/azure/frontdoor/frontDoorWafDefaultRuleSet.js:50]:
```suggestion
                if (ruleSetVersion >= 2.1) {
```
make the ruleSetVersion a constant at the top so it makes it easier for periodic refactoring, also add it in the sheet

File: plugins/azure/frontdoor/frontDoorWafDefaultRuleSet.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                    } else {
+                        helpers.addResult(results, 2, `Front Door WAF policy has latest ${ruleSet.ruleSetType}: ${ruleSet.ruleSetVersion} default rule set configured with ${ruleSetAction} action`, location, policy.id);
+                    }
+                } else {
+                    helpers.addResult(results, 2, 'Front Door WAF policy does not have latest default rule set configured', location, policy.id);
+                }
+            }
+            
+            if (!premiumWAF) {
+                helpers.addResult(results, 0, 'No existing Front Door WAF policies found', location);
```
Comment (mehakseedat63) [inline plugins/azure/frontdoor/frontDoorWafDefaultRuleSet.js:62]:
```suggestion
                helpers.addResult(results, 0, 'No existing Front Door WAF policies found', location);
```
this message should either say that no premium policies found or we should produce a pass result for non-premium policies

File: plugins/azure/frontdoor/frontDoorWafDefaultRuleSet.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Front Door WAF Latest Default Rule Set',
+    category: 'Front Door',
+    domain: 'Content Delivery',
+    severity: 'Low',
+    description: 'Ensures that Azure Front Door WAF is using the latest Azure managed default rule set with action set to block.',
+    more_info: 'Azure-managed rule sets provide an easy way to deploy protection against a common set of security threats. Azure updates these rules as needed to protect against new attack signatures. ',
+    recommended_action: 'Modify the Front Door WAF policy and add latest default rule set.',
```
Comment (mehakseedat63) [inline plugins/azure/frontdoor/frontDoorWafDefaultRuleSet.js:11]:
```suggestion
    description: 'Ensures that Azure Front Door WAF is using the latest Azure managed default rule set with action set to block.',
    more_info: 'Azure-managed rule sets provide an easy way to deploy protection against a common set of security threats. Azure updates these rules as needed to protect against new attack signatures. ',
    recommended_action: 'Modify the Front Door WAF policy and add latest default rule set.',
```
should mention that this is only for premium policies

File: plugins/azure/frontdoor/frontDoorWafDefaultRuleSet.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                var ruleSetAction = ruleSet.ruleSetAction ? ruleSet.ruleSetAction.toLowerCase() : '';
+
+                if (ruleSetVersion >= 2.1) {
+                    if (ruleSetAction == 'block') {
+                        helpers.addResult(results, 0, `Front Door WAF policy has latest ${ruleSet.ruleSetType}: ${ruleSet.ruleSetVersion} default rule set configured with ${ruleSetAction} action`, location, policy.id);
+                    } else {
+                        helpers.addResult(results, 2, `Front Door WAF policy has latest ${ruleSet.ruleSetType}: ${ruleSet.ruleSetVersion} default rule set configured with ${ruleSetAction} action`, location, policy.id);
+                    }
+                } else {
+                    helpers.addResult(results, 2, 'Front Door WAF policy does not have latest default rule set configured', location, policy.id);
```
Comment (mehakseedat63) [inline plugins/azure/frontdoor/frontDoorWafDefaultRuleSet.js:57]: this will fail if we don't find a default rule set?

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

File: plugins/azure/frontdoor/frontDoorWafDefaultRuleSet.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Front Door WAF Latest Default Rule Set',
+    category: 'Front Door',
+    domain: 'Content Delivery',
+    severity: 'Low',
+    description: 'Ensures that Azure Front Door WAF is using the latest Azure managed default rule set with action set to block.',
```
Comment (alphadev4) [inline plugins/azure/frontdoor/frontDoorWafDefaultRuleSet.js:9]:
```suggestion
    description: 'Ensures that Azure Front Door Premium WAF is using the latest Azure managed default rule set with action set to block.',
```

File: plugins/azure/frontdoor/frontDoorWafDefaultRuleSet.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    more_info: 'Azure-managed rule sets provide an easy way to deploy protection against a common set of security threats. Azure updates these rules as needed to protect against new attack signatures. ',
```
Comment (alphadev4) [inline plugins/azure/frontdoor/frontDoorWafDefaultRuleSet.js:10]:
```suggestion
    more_info: 'Azure-managed rule sets provide an easy way to deploy protection against a common set of security threats. Azure updates these rules as needed to protect against new attack signatures.',
```

File: plugins/azure/frontdoor/frontDoorWafDefaultRuleSet.js (hunk 3 above)
Comment (alphadev4) [inline plugins/azure/frontdoor/frontDoorWafDefaultRuleSet.js:11]:
```suggestion
    recommended_action: 'Modify the Front Door Premium WAF policy and add latest default rule set.',
```

File: plugins/azure/frontdoor/frontDoorWafDefaultRuleSet.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    recommended_action: 'Modify the Front Door WAF policy and add latest default rule set.',
+    link: 'https://learn.microsoft.com/en-us/azure/web-application-firewall/afds/afds-overview#waf-policy-and-rules',
+    apis: ['afdWafPolicies:listAll'],
+    realtime_triggers: ['microsoftnetwork:frontdoorwebapplicationfirewallpolicies:write', 'microsoftnetwork:frontdoorwebapplicationfirewallpolicies:delete'],
+
+    run: function (cache, settings, callback) {
```
Comment (alphadev4) [inline plugins/azure/frontdoor/frontDoorWafDefaultRuleSet.js:16]:
```suggestion
    run: function(cache, settings, callback) {
```

**Review by mehakseedat63 — APPROVED** [review]

**Pull request merged**